src/main/groovy/org/jetbrains/intellij/tasks/RunPluginVerifierTask.groovy

```diff
@@ -0,0 +1,354 @@
+        this.pluginsToCheckFile = pluginsToCheckFile
+    }
+
+    @Input
+    @Optional
+    String getSubsystemsToCheck() {
+        return subsystemsToCheck
+    }
+
+    void setSubsystemsToCheck(String subsystemsToCheck) {
+        this.subsystemsToCheck = subsystemsToCheck
+    }
+
+    @TaskAction
+    void runPluginVerifier() {
+        def extension = project.extensions.findByType(IntelliJPluginExtension)
+        def pluginFileName = "${extension.pluginName}-${project.version}"
+
+        if (!project.file("${project.buildDir}/distributions/${pluginFileName}.zip").exists()) {
+            throw new IllegalStateException("Plugin file $pluginFileName does not exist.")
+        }
+
+        if (ides.isEmpty()) {
+            ides.add("${extension.type}-${extension.version}")
+        }
+
+        def resolver = new IdeaDependencyManager(extension.intellijRepo ?: IntelliJPlugin.DEFAULT_INTELLIJ_REPO)
+        def verifierPath = getVerifierPath()
+
+        def verifierArgs = ["check-plugin"]
+        verifierArgs += getOptions()
+        verifierArgs += ["${project.buildDir}/distributions/${pluginFileName}.zip"]
+        verifierArgs += ides.collect {
+            def (String type, String version) = it.split("-")
+            def dependency = resolver.resolveRemote(project, version, type, false)
+            return dependency.classes.absolutePath
+        }
+
+        new ByteArrayOutputStream().withStream { os ->
+            project.javaexec {
+                classpath = project.files(verifierPath)
+                main = "com.jetbrains.pluginverifier.PluginVerifierMain"
+                args = verifierArgs
+                standardOutput = os
+            }
+
+            def output = os.toString()
+            println output
+            for (FailureLevel level : FailureLevel.values()) {
+                if (failureLevel.contains(level) && output.contains(level.testValue)) {
+                    throw new GradleException(level.toString())
+                }
+            }
+        }
+    }
+
+    private String getVerifierPath() {
+        def repository = project.repositories.maven { it.url = IntelliJPlugin.DEFAULT_INTELLIJ_PLUGIN_SERVICE }
+        try {
+            def resolvedVerifierVersion = resolveVerifierVersion()
+            def dependency = project.dependencies.create("org.jetbrains.intellij.plugins:verifier-cli:$resolvedVerifierVersion:all@jar")
+            def configuration = project.configurations.detachedConfiguration(dependency)
+            return configuration.singleFile.absolutePath
+        }
+        finally {
+            project.repositories.remove(repository)
+        }
+    }
+
+    private String resolveVerifierVersion() {
+        if (verifierVersion != VERIFIER_VERSION_LATEST) {
+            return verifierVersion
+        }
+
+        def url = new URL(BINTRAY_API_VERIFIER_VERSION_LATEST)
+        return new JsonSlurper().parse(url)["name"]
+    }
+
+    private String resolveRuntimeDir() {
```

Comment:
looks too similar to org.jetbrains.intellij.IntelliJPlugin#prepareConventionMappingsForRunIdeTask

Reply:
Indeed, but I'm resolving different paths here - not to the binary but JRE home dir. For `builtinJbrVersion` case it's more _complicated_ than `builtinJbrVersion` in the mentioned method.